core/contracts/financial-templates/implementation/FeePayer.sol

```diff
@@ -157,4 +157,13 @@ contract FeePayer is Testable {
         FixedPoint.Unsigned memory adjustedCollateral = collateralToAdd.div(cumulativeFeeMultiplier);
         rawCollateral.rawValue = rawCollateral.add(adjustedCollateral).rawValue;
     }
+
+    // Effectively performs the opposite operation from _getCollateral().
+    function _convertCollateral(FixedPoint.Unsigned memory collateral)
+        internal
+        view
+        returns (FixedPoint.Unsigned memory rawCollateral)
+    {
+        return collateral.div(cumulativeFeeMultiplier);
```

Comment:
As discussed IRL, You can call `_convertCollateral` in `_addCollateral` and `_subtractCollateral`.

Perhaps some more readable ways to name `_convertCollateral` and `_getCollateral` would be `_scaledUp` and `scaledDown` collateral. Maybe `_snapshot` and `_apply` would work, but `_apply` is arguably less readable than `_getCollateral`.
- `PreFeeCollateral` and `PostFeeCollateral`?

What we're trying to convey is that one method (`_convertCollateral`) artificially increases the value of 1 collateral unit to account for the fact that fees have already been extracted from the contract but they shouldn't affect that ` collateral unit today. Conversely, the other (`_getCollateral`) decreases the value of 1 collateral unit to account for the fact that it has been reduced by some amount to pay for fees **since the collateral was originally put in**

Reply:
Hmmm, I worry that pre and post imply some sort of time relationship, which I feel is a little different than what we're trying to communicate.

Please let me know if you disagree, though.

Incorporated the first part of your comment -- PTAL.

Reply:
makes sense. Comments above _getCollateral and _convertCollateral would make it more readable without having to change names